protocol/communities/manager_torrent.go

```diff
@@ -0,0 +1,675 @@
+//go:build !disable_torrent
+// +build !disable_torrent
+
+// Attribution to Pascal Precht, for further context please view the below issues
+// - https://github.com/status-im/status-go/issues/2563
+// - https://github.com/status-im/status-go/issues/2565
+// - https://github.com/status-im/status-go/issues/2567
+// - https://github.com/status-im/status-go/issues/2568
+
+package communities
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/protocol/encryption"
+	"github.com/status-im/status-go/protocol/transport"
+	"github.com/status-im/status-go/signal"
+
+	"github.com/anacrolix/torrent"
+	"github.com/anacrolix/torrent/metainfo"
+	"go.uber.org/zap"
+)
+
+type archiveMDSlice []*archiveMetadata
+
+type archiveMetadata struct {
+	hash string
+	from uint64
+}
+
+func (md archiveMDSlice) Len() int {
+	return len(md)
+}
+
+func (md archiveMDSlice) Swap(i, j int) {
+	md[i], md[j] = md[j], md[i]
+}
+
+func (md archiveMDSlice) Less(i, j int) bool {
+	return md[i].from > md[j].from
+}
+
+type EncodedArchiveData struct {
+	padding int
+	bytes   []byte
+}
+
+type TorrentManager struct {
+	torrentConfig                *params.TorrentConfig
+	torrentClient                *torrent.Client
+	torrentTasks                 map[string]metainfo.Hash
+	historyArchiveDownloadTasks  map[string]*HistoryArchiveDownloadTask
+	historyArchiveTasksWaitGroup sync.WaitGroup
+	historyArchiveTasks          sync.Map // stores `chan struct{}`
+
+	logger       *zap.Logger
+	stdoutLogger *zap.Logger
```

Comment:
Same thing about `stdoutLogger`. Feel free to kick it.

Reply:
https://github.com/status-im/status-go/pull/5257/commits/29d7c3f7ef82f5b57e67c220c08575ab32d90a86